jax/interpreters/partial_eval.py

```diff
@@ -423,20 +423,31 @@ def tracers_to_jaxpr(in_tracers, out_tracers):
     `invars`.
   """
   newvar = core.gensym('')
-  t_to_var = defaultdict(newvar)
-  getvar = lambda t: t_to_var[id(t)]
+  t_to_var = {}
+  def getvar(t):
+    var = t_to_var.get(id(t))
+    if var is None:
+      var = newvar(partial_val_aval(*t.pval))
+      t_to_var[id(t)] = var
+    return var
   sorted_tracers = toposort(out_tracers)
   invars = map(getvar, in_tracers)
   eqns = []
   env = {}
   consts = {}
-  const_to_var = defaultdict(newvar)
+  const_to_var = {}
+  def getconstvar(c):
+    var = const_to_var.get(id(c))
```

Comment:
I think that the previous code would hash on the id of the constant value, not the constant, which means it would reuse the same var for the same constant?

Reply:
Fixed